dotnet/src/Plugins/Plugins.Web/Google/GoogleTextSearch.cs

```diff
@@ -0,0 +1,307 @@
+
+            foreach (var filterClause in filterClauses)
+            {
+                if (filterClause is EqualityFilterClause equalityFilterClause)
+                {
+                    var value = equalityFilterClause.Value as string;
+                    if (value is null)
+                    {
+                        continue;
+                    }
+
+                    if (equalityFilterClause.FieldName.Equals("cr", StringComparison.OrdinalIgnoreCase))
+                    {
+                        search.Cr = value;
+                    }
+                    else if (equalityFilterClause.FieldName.Equals("dateRestrict", StringComparison.OrdinalIgnoreCase))
+                    {
+                        search.DateRestrict = value;
+                    }
+                    else if (equalityFilterClause.FieldName.Equals("exactTerms", StringComparison.OrdinalIgnoreCase))
+                    {
+                        search.ExactTerms = value;
+                    }
+                    else if (equalityFilterClause.FieldName.Equals("excludeTerms", StringComparison.OrdinalIgnoreCase))
+                    {
+                        search.ExcludeTerms = value;
+                    }
+                    else if (equalityFilterClause.FieldName.Equals("filter", StringComparison.OrdinalIgnoreCase))
+                    {
+                        search.Filter = value;
+                    }
+                    else if (equalityFilterClause.FieldName.Equals("gl", StringComparison.OrdinalIgnoreCase))
+                    {
+                        search.Gl = value;
+                    }
+                    else if (equalityFilterClause.FieldName.Equals("hl", StringComparison.OrdinalIgnoreCase))
+                    {
+                        search.Hl = value;
+                    }
+                    else if (equalityFilterClause.FieldName.Equals("linkSite", StringComparison.OrdinalIgnoreCase))
+                    {
+                        search.LinkSite = value;
+                    }
+                    else if (equalityFilterClause.FieldName.Equals("lr", StringComparison.OrdinalIgnoreCase))
+                    {
+                        search.Lr = value;
+                    }
+                    else if (equalityFilterClause.FieldName.Equals("orTerms", StringComparison.OrdinalIgnoreCase))
+                    {
+                        search.OrTerms = value;
+                    }
+                    else if (equalityFilterClause.FieldName.Equals("rights", StringComparison.OrdinalIgnoreCase))
+                    {
+                        search.Rights = value;
+                    }
+                    else if (equalityFilterClause.FieldName.Equals("siteSearch", StringComparison.OrdinalIgnoreCase))
+                    {
+                        search.SiteSearch = value;
+                        search.SiteSearchFilter = CseResource.ListRequest.SiteSearchFilterEnum.I;
+                    }
```

Comment:
For each if I would check first if the `search.XYZ` is set, preventing the `Equals` check that might be costly in this foreach loop.

Reply:
The code has this check

```csharp
if (equalityFilterClause.Value is not string value)
{
    continue;
}
```

Reply:
see below what I meant:
```csharp
if (search.Cr is not null && equalityFilterClause.FieldName.Equals("cr", StringComparison.OrdinalIgnoreCase))
{
    search.Cr = value;
}
```

Reply:
I added a delegate and a dictionary to look up the appropriate setter.